index.js

```diff
@@ -0,0 +1,133 @@
+/*
+        @@@@@@@            @@@@@@@         @@
+      @@       @@        @@       @@      @@@
+    @@   @@@ @@  @@    @@   @@@ @@  @@   @@@@@@ @@   @@@  @@ @@@      @@@@
+   @@  @@   @@@   @@  @@  @@   @@@   @@   @@@   @@   @@@  @@@   @@  @@@   @@
+   @@  @@   @@@   @@  @@  @@   @@@   @@   @@@   @@   @@@  @@@   @@  @@@@@@@@
+   @@  @@   @@@  @@   @@  @@   @@@  @@    @@@   @@   @@@  @@@   @@  @@@
+    @@   @@@ @@@@@     @@   @@@ @@@@@      @@@    @@@ @@  @@@@@@      @@@@@
+      @@                 @@                           @@  @@
+        @@@@@@@            @@@@@@@               @@@@@    @@
+                                                          */
+//. # sanctuary-type-identifiers
+//.
+//. A type is a set of values. The Boolean type, for example, is the type
+//. comprising `true` and `false`. A value may be a member of several types.
+//. `42`, for example, is a member of Number, PositiveNumber, Integer,
+//. PositiveInteger, and many other types.
+//.
+//. In certain situations it is useful to divide JavaScript values into
+//. non-overlapping types. The language provides two constructs to support
+//. this: the [`typeof`][1] operator, and [`Object.prototype.toString`][2].
+//. Each has pros and cons, but neither supports user-defined types.
+//.
+//. This package specifies an [algorithm][3] for deriving a type identifier
+//. from any JavaScript value, and exports an implementation of this algorithm.
+//. Authors of algebraic data types may follow this specification in order to
+//. make their data types compatible with the algorithm.
+//.
+//. ### Algorithm
+//.
+//. 1.  Take any JavaScript value, `x`.
+//.
+//. 2.  If `x` is `null` or `undefined`, go to step 6.
+//.
+//. 3.  If the value of `x.constructor` is `null` or `undefined`, go to step 6.
+//.
+//. 4.  If the value of `x.constructor.prototype` is identical to `x`, go to
+//.     step 6. (This check prevents a prototype object from being treated as
+//.     a member of the type with which it is associated.)
+//.
+//. 5.  If `typeof x.constructor['@@type']` evaluates to `'string'`, return
+//.     the value of `x.constructor['@@type']`; otherwise, go to step 6.
+//.
+//. 6.  Return the [`Object.prototype.toString`][2] representation of `x`
+//.     without the leading `'[object '` and trailing `']'`.
+//.
+//. ### Compatibility
+//.
+//. For an algebraic data type to be compatible with the [algorithm][3], the
+//. following must be true:
+//.
+//.   - every member of the type must have a `constructor` property pointing
+//.     to an object known as a _type representative_;
+//.
+//.   - the type representative must have a `@@type` property; and
+//.
+//.   - the type representative's `@@type` property must be a type identifier,
+//.     a string primitive in the form `'<package-name>/<type-name>'`, where
```

Comment:
The form is not quite must I think, so maybe we should use "recommended" for that?
```diff
-a string primitive in the form 
+a string primitive. it's recommended to be in the form
```